### PR TITLE
Fix bug, if alpaka_add_library() and Clang as CUDA compiler is used together

### LINKED ISSUE #1359: CMake bug alpaka_add_library(${_TARGET} STATIC) with Clang as CUDA compiler
If I use `alpaka_add_library(${TARGET_NAME} STATIC  file.cpp)` and compile my project with Clang as CUDA compiler, I get following error:

```bash
-- Configuring done
CMake Error at alpaka/cmake/addLibrary.cmake:95 (ADD_LIBRARY):
  Cannot find source file:
    STATIC
  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .h .hh .h++
  .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .ispc
Call Stack (most recent call first):
  cmake/cuplaTargetHelper.cmake:27 (alpaka_add_library)
  CMakeLists.txt:101 (createCuplaTarget)
CMake Error at alpaka/cmake/addLibrary.cmake:95 (ADD_LIBRARY):
  No SOURCES given to target: cupla
Call Stack (most recent call first):
  cmake/cuplaTargetHelper.cmake:27 (alpaka_add_library)
  CMakeLists.txt:101 (createCuplaTarget)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```
Full log: https://gitlab.com/hzdr/crp/cupla/-/jobs/1422713016

The problem is the following lines of code: https://github.com/alpaka-group/alpaka/blob/394eb4e5eea81b1a418bd33f893f53c97c1e77bb/cmake/addLibrary.cmake#L95-L101

`libaryType` should be on the second position. I expected, that this bug only affected alpaka `0.6.1` and older, because we replaced the affected code in alpaka `0.7.0`.

### PR DESCRIPTION
**Pull request by SimeonEhrig**
I tested the code with the modernized cmake for cupla: https://github.com/alpaka-group/cupla/pull/203
It uses alpaka_add_library, if the examples are build.

It's related to the issue #1359

### CODE REVIEW

**Review by j-stephan — DISMISSED** [review]

**Comment by j-stephan** [comment]
@alpaka-group/alpaka-maintainers 

Since this PR has no problems we could merge this. However, this change only affects our `0.6.1` branch (and older versions which we don't care about) because the way this is handled has changed with `0.7.0`.

1. Do we want to incorporate this PR into 0.6.1? The issue is rather exotic since nobody has found it before and I doubt many people care about this combination of compilers + options.
2. If we want to merge the PR, do we need to bump the version number to 0.6.2?

Until we have found a consensus here please DO NOT MERGE this PR.

**Comment by psychocoderHPC** [comment]
Wait 0.6.1 branch should be deleted, I forget to delete this branch. The release is already out and merging something into the 0.6.1 branch will have no effect.

**Comment by psychocoderHPC** [comment]
I redirected this PR against 0.6.2 and opened also a branch 0.6.2, removed the 0.6.1 milestone and opened a project 0.6.2

**Comment by psychocoderHPC** [comment]
@j-stephan This is broken in 0.7.0 too and must be backported on Monday!

**Comment by psychocoderHPC** [comment]
@SimeonEhrig could it be that the order of arguments is wrong for HIP too?

File: cmake/addLibrary.cmake
Comment (psychocoderHPC) [inline cmake/addLibrary.cmake]: PLease move source files as last argument to fulfill the Cmake description https://cmake.org/cmake/help/latest/command/add_library.html

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

**Comment by SimeonEhrig** [comment]
> @SimeonEhrig could it be that the order of arguments is wrong for HIP too?

@psychocoderHPC Looks like so. I will test it. But if HIP is also affected, we need to fix it in `alpaka 0.6.2` and `0.7.0`. Should I open a second PR or do you pull somehow to the current dev branch and backport it? Could be complicated because the cmake was strong changed since `0.6.1`.

**Comment by j-stephan** [comment]
>  But if HIP is also affected, we need to fix it in alpaka 0.6.2 and 0.7.0. Should I open a second PR or do you pull somehow to the current dev branch and backport it? Could be complicated because the cmake was strong changed since 0.6.1.

Why would we need a second PR? Can't this be changed as part of this PR, too?

**Comment by j-stephan** [comment]
Please also update the author list and copyright year!

**Comment by SimeonEhrig** [comment]
> > But if HIP is also affected, we need to fix it in alpaka 0.6.2 and 0.7.0. Should I open a second PR or do you pull somehow to the current dev branch and backport it? Could be complicated because the cmake was strong changed since 0.6.1.
> 
> Why would we need a second PR? Can't this be changed as part of this PR, too?

Okay, it became a little bit tricky. To make it short. This bug does not affect HIP, because the API of `hip_add_library` is different to `add_library` but @psychocoderHPC recognized correctly, that this bug also affected version `0.7.0`. To not slow down the release of alpaka `0.7.0`, @psychocoderHPC suggested that I open extra PRs for the `develop` and the `0.7.0-rc1` branch.

**Review by j-stephan — APPROVED** [review]

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**
